app/services/immediate_email_generation_service.rb

```diff
@@ -17,13 +17,20 @@ def call
           queue: content.queue,
         )
       end
+      update_last_alerted_at(batch)
     end
   end
 
 private
 
   attr_reader :content
 
+  def update_last_alerted_at(batch)
```

Comment:
The problem with doing it here is that if you have a subscriber list with 20,000 subscribers, the record is going to get updated 20,000 times.